### ISSUE TITLE
Order of plugins affects output

### ISSUE THREAD

**Issue by CausticYarn** [marshmallow-code/apispec#136]
The example shows:

```
# Create an APISpec
spec = APISpec(
    title='Swagger Petstore',
    version='1.0.0',
    plugins=[
        'apispec.ext.flask',
        'apispec.ext.marshmallow',
    ],
)
```
 
And the output is:

```
#         "responses": {
#           "200": {
#             "schema": {
#               "$ref": "#/definitions/Pet"
#             },
#             "description": "A pet to be returned"
#           }
```

However, if the order of the plugins to spec is (marshmallow, flask), then the output is:

```
#         "responses": {
#           "200": {
#             "schema": "PetSchema"
#             "description": "A pet to be returned"
#           }
```

Which doesn't resolve correctly in the Swagger doc.  If the order is relevant by design, it should be documented.

**Comment by sloria** [comment]
This looks to be a bug; order should not matter, at least not for the flask and marshmallow plugins. I would definitely review and merge a PR fixing this.

**Comment by yoichi** [comment]
Order dependency comes from the fact that operation resolve process is done inside path helpers.
It can be solved by separating them https://github.com/marshmallow-code/apispec/pull/152

We have to introduce incompatible change to extension mechanism to solve it. I'm not sure we should fix this issue or not. What do you think? @sloria

**Comment by sloria** [comment]
This is fixed in 0.27.0
